runtime/cudaq/platform/fermioniq/helpers/FermioniqServerHelper.cpp

```diff
@@ -0,0 +1,512 @@
+  for (auto &[result, info] : output_names) {
+    cudaq::info("Qubit {} Name {}", info.qubitNum,
+                info.registerName);
+  }
+  */
+
+  RestClient client;
+
+  refreshTokens(false);
+
+  auto headers = getHeaders();
+
+  std::string path = "/api/jobs/" + jobID + "/results";
+
+  auto response_json = client.get(backendConfig.at(CFG_URL_KEY), path, headers);
+
+  cudaq::debug("got job result: {}", response_json.dump());
+
+  auto metadata = response_json.at("metadata");
+  // cudaq::info("metadata: {}", metadata.dump());
+  auto output = response_json.at("emulator_output");
+
+  cudaq::sample_result sample_result;
+
+  // to-do: restrict to 1 circuit
+  for (const auto &it : output.items()) {
+    // cudaq::info("result: {}", it.value().dump());
+    // int circuit_number = it.value().at("circuit_number");
+
+    // "samples":{"00000":500,"11111":500}
+    auto output = it.value().at("output");
+    auto samples = output.at("samples");
+
+    CountsDictionary sample_dict;
+    for (const auto &[qubit_str, n_observed] : samples.items()) {
+      sample_dict[qubit_str] = n_observed;
+    }
+
+    if (output.contains("expectation_values")) {
+      auto exp_vals = output.at("expectation_values");
+
+      double exp = exp_vals[0].at("expval").at("real");
+
+      ExecutionResult exec_result(sample_dict, exp);
+      sample_result = cudaq::sample_result(exec_result);
+    } else {
+      ExecutionResult exec_result(sample_dict);
+      sample_result = cudaq::sample_result(exec_result);
+    }
+
+    break;
+  }
+
+#if 0
+  // First reorder the global register by QIR qubit number.
+  std::vector<std::size_t> qirQubitMeasurements;
+  qirQubitMeasurements.reserve(output_names.size());
+  for (auto &[result, info] : output_names)
+    qirQubitMeasurements.push_back(info.qubitNum);
+
+  std::vector<std::size_t> idx(output_names.size());
+  std::iota(idx.begin(), idx.end(), 0);
+  std::sort(idx.begin(), idx.end(), [&](std::size_t i1, std::size_t i2) {
+    return qirQubitMeasurements[i1] < qirQubitMeasurements[i2];
+  });
+  cudaq::info("Reordering global result to map QIR result order to QIR qubit "
+              "allocation order is {}",
+              idx);
+  sample_result.reorder(idx);
+
+  // Now reorder according to reorderIdx[]. This sorts the global bitstring in
+  // original user qubit allocation order.
+  auto thisJobReorderIdxIt = reorderIdx.find(jobID);
+  if (thisJobReorderIdxIt != reorderIdx.end()) {
+    auto &thisJobReorderIdx = thisJobReorderIdxIt->second;
+    if (!thisJobReorderIdx.empty())
+      sample_result.reorder(thisJobReorderIdx);
+  }
+
+  // We will also make registers for each result using output_names.
+  for (auto &[result, info] : output_names) {
+    cudaq::sample_result singleBitResult = sample_result.get_marginal({result});
+    ExecutionResult executionResult{singleBitResult.to_map(),
+                                    info.registerName};
+    sample_result.append(executionResult);
+  }
+#endif
```

Comment:
We can just remove this block if it's not relevant.

Reply:
done